src/components/shared/RecentJobsList.vue

```diff
@@ -1,29 +1,47 @@
 <script>
 import { mapState, mapActions } from 'vuex';
 import JobListing from './JobListing';
+import JobPeriodSelector from './JobPeriodSelector';
+import { PERIODS } from '../../store/constants';
 
 export default {
   components: {
     JobListing,
+    JobPeriodSelector,
   },
   data() {
     return {
       isLoading: false,
     };
   },
   computed: {
-    ...mapState(['recentPosts']),
+    ...mapState(['recentPosts', 'activePeriod']),
   },
   methods: {
-    ...mapActions(['fetchRecentPosts']),
+    ...mapActions(['fetchRecentPosts', 'setPeriod']),
+    fetch() {
+      this.isLoading = true;
+
+      this.fetchRecentPosts()
+        .then(() => {
+          this.isLoading = false;
+        });
+    },
+    handlerSetPeriod() {
```

Comment:
We can change the method name to `handleSetPeriod`. If you want to use `handler` then it should be at the at `setPeriodHandler`.